src/main/java/org/truffleruby/core/adapters/OutputStreamAdapter.java

```diff
@@ -35,8 +34,7 @@ public OutputStreamAdapter(RubyContext context, DynamicObject object, Encoding e
     @Override
     public void write(int bite) throws IOException {
         context.send(object, "write", null,
-                StringOperations.createString(context, RopeOperations.ropeFromByteList(
-                        RopeBuilder.createRopeBuilder(new byte[]{(byte) bite}, encoding), CodeRange.CR_VALID)));
+                StringOperations.createString(context, RopeBuilder.createRopeBuilder(new byte[]{(byte) bite}, encoding).toRope(CodeRange.CR_VALID)));
```

Comment:
This sounds fiarly inefficient, but, out of scope of this PR.
(Could maybe use RopeConstants or `putc`)

Reply:
I have a fix for that already. I'll open a new PR after this is merged.